common/path.go

```diff
@@ -100,14 +100,28 @@ func DefaultAssetPath() string {
 }
 
 func DefaultDataDir() string {
-	usr, _ := user.Current()
-	if runtime.GOOS == "darwin" {
-		return filepath.Join(usr.HomeDir, "Library", "Ethereum")
-	} else if runtime.GOOS == "windows" {
-		return filepath.Join(usr.HomeDir, "AppData", "Roaming", "Ethereum")
-	} else {
-		return filepath.Join(usr.HomeDir, ".ethereum")
+	// Try to place the data folder in the user's home dir
+	if usr, err := user.Current(); err == nil {
+		if runtime.GOOS == "darwin" {
+			return filepath.Join(usr.HomeDir, "Library", "Ethereum")
+		} else if runtime.GOOS == "windows" {
+			return filepath.Join(usr.HomeDir, "AppData", "Roaming", "Ethereum")
+		} else {
+			return filepath.Join(usr.HomeDir, ".ethereum")
+		}
+	}
+	// If the user cannot be retrieved (Android, Fedora/Go bug), use the current folder
+	if dir, err := os.Getwd(); err == nil {
+		if runtime.GOOS == "darwin" {
+			return filepath.Join(dir, "Ethereum")
+		} else if runtime.GOOS == "windows" {
+			return filepath.Join(dir, "Ethereum")
+		} else {
```

Comment:
Well, anoth eroption would be to drop the second block (the pwd one), and check in geth main that an empty data dir should return an error. I'm fine with that approach too. This is actually why I never implemented this, because I couldn't decide what the best course of action is.